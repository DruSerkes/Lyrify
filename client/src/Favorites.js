import React from 'react';
// import { useDispatch, shallowEqual } from 'react-redux';
// import { getFavorites } from './reducers/actions';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Favorite from './Favorite';

const Favorites = ({ user, favorites }) => {
	// const dispatch = useDispatch();
	// // const favorites = useSelector((state) => state.favorites, shallowEqual);

	// useEffect(
	// 	() => {
	// 		if (!favorites.length) dispatch(getFavorites(user.id));
	// 	},
	// 	[ favorites, dispatch, user.id ]
	// );

	return (
		<React.Fragment>
			<Typography variant="h1">Favorites</Typography>
			<List className="Favorites">
				{favorites.length ? (
					favorites.map((fav) => (
						<Favorite artist={fav.artist} song={fav.song} id={fav.id} key={fav.id} userId={user.id} />
					))
				) : (
					<Typography>You haven't favorited any lyrics!</Typography>
				)}
			</List>
		</React.Fragment>
	);
};

export default Favorites;
