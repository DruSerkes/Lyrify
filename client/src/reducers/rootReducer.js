import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
