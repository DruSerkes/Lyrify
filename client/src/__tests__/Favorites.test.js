import React from 'react';
import { render } from '@testing-library/react';
import Favorites from '../Favorites';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Favorites favorites={[ { id: 1, artist: 'test', song: 'testsong' } ]} user={{ id: 2 }} />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Favorites favorites={[ { id: 1, artist: 'test', song: 'testsong' } ]} user={{ id: 2 }} />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
