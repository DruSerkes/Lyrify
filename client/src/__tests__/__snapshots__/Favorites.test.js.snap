// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Favorites
      </h1>
    </div>
    <ul
      class="MuiList-root Favorites MuiList-padding"
    >
      <li
        class="MuiListItem-root Favorite MuiListItem-gutters MuiListItem-divider"
        id="1"
      >
        <a
          class="Favorite-Item"
          href="/favorites/1"
        >
          test: "testsong"
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Remove
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
